File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
function solution(s, n) {
    let answer = '';

    for (let i = 0; i < s.length; i++) {
        let ch = s[i];
        if (ch === ' ') {
            answer += ' '; // 공백은 그대로
        } else {
            let code = s.charCodeAt(i);

            if (code >= 65 && code <= 90) {
                // 대문자 A-Z
                answer += String.fromCharCode((code - 65 + n) % 26 + 65);
            } else if (code >= 97 && code <= 122) {
                // 소문자 a-z
                answer += String.fromCharCode((code - 97 + n) % 26 + 97);
            }
        }
    }

    return answer;
}

File: 프로그래머스/1/12926. 시저 암호/시저 암호.py
def solution(s, n):
    answer = ''
    for ch in s:
        if ch == ' ':
            answer += ' '   # 공백은 그대로
        elif 'A' <= ch <= 'Z':
            answer += chr((ord(ch) - ord('A') + n) % 26 + ord('A'))
        elif 'a' <= ch <= 'z':
            answer += chr((ord(ch) - ord('a') + n) % 26 + ord('a'))
    return answer
